ServerlessLibraryAPI/ClientApp/src/components/Contribute/AddContributionForm.js

```diff
@@ -71,47 +75,61 @@ class AddContribution extends Component {
   }
 
   render() {
-    const { showForm, error } = this.state;
+    const { showForm } = this.state;
     return (
-      <div>
-        <Link onClick={this.onLinkClick}>Add new contribution</Link>
-        {error && error !== "" && (
-          <div>Submission failed! Details: {error}</div>
-        )}
+      <div className="add-contribution-container">
+        <div className="add-contribution-link">
+          <Link onClick={this.onLinkClick}>
+            <Icon iconName="Edit" style={{ marginRight: "5px" }} />
+            Add new contribution
+          </Link>
+        </div>
         {showForm && (
-          <div>
```

Comment:
This can be a form control. In that case you don't need to use local states for each fields

Reply:
even with forms, as https://reactjs.org/docs/forms.html we should still controlled components for input fields.